src/Dapr.Actors/Client/ActorProxyFactory.cs

```diff
@@ -6,58 +6,56 @@
 namespace Dapr.Actors.Client
 {
     using System;
+    using System.Text.Json;
     using Dapr.Actors.Builder;
     using Dapr.Actors.Communication.Client;
 
     /// <summary>
     /// Represents a factory class to create a proxy to the remote actor objects.
     /// </summary>
-    internal class ActorProxyFactory : IActorProxyFactory
+    public class ActorProxyFactory : IActorProxyFactory
     {
         private readonly IDaprInteractor daprInteractor;
 
+        /// <summary>
+        /// The <see cref="JsonSerializerOptions"/> used for actor message serialization for remoting
+        /// </summary>
+        public readonly JsonSerializerOptions JsonSerializerOptions = new JsonSerializerOptions();
```

Comment:
We should use `JsonSerializerDefaults.Web` or at least the equivalent settings on .NET 5 on .NET 3.1.  https://github.com/dotnet/runtime/blob/1d9e50cb4735df46d3de0cee5791e97295eaf588/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs#L103

Reply:
Nice, I wasn't aware of this new default. I don't see any equivalent on .NET 3.1 - would you settle for a static class? 
``` C#
namespace Dapr.Actors.Common
{
    /// <summary>
    /// A helper used to stanardize the <see cref="JsonSerializerOptions"/> defaults
    /// </summary>
    public static class JsonSerializerDefaults
    {
        /// <summary>
        /// The <see cref="JsonSerializerOptions"/> used for actor message serialization
        /// </summary>
        public static JsonSerializerOptions Web = new JsonSerializerOptions()
            { 
                PropertyNamingPolicy = JsonNamingPolicy.CamelCase, 
                PropertyNameCaseInsensitive = true
            };
    }
}
```

Used like:
``` C#
public JsonSerializerOptions JsonSerializerOptions { get; set; } = JsonSerializerDefaults.Web;
```

The only issue is there isn't any common shared project across all these Dapr projects so I'd either have to make a new one just for this class or I'd have to duplicate it in each project.

Reply:
I have moved the `JsonSerializerOptions`  to the `ActorProxyOptions` class which is used by the `ActorProxyFactory` to initialize the `ActorProxy`s. I have also set the default `ActorProxy`'s `JsonSerializerOptions` to the new static instance I made `JsonSerializerDefaults.Web`. This seems to simplify it a bit more than what I had originally.

Reply:
On second thought - a mutable static instance for `JsonSerializerOptions` isn't a good idea. I've change it to a static method that returns a new `JsonSerializerOptions` instance and will leave it up to the user to reuse it.

Reply:
Thanks @lukepammant - I'll take a look. Agreed, it's not good to expose a static mutable thing.